inbox/mailsync/backends/imap/generic.py

```diff
@@ -517,8 +517,13 @@ def resync_uids_impl(self):
         with session_scope(self.namespace_id) as db_session:
             invalid_uids = {
                 uid
-                for uid, in db_session.query(ImapUid.msg_uid).filter_by(
-                    account_id=self.account_id, folder_id=self.folder_id
+                for uid, in db_session.query(ImapUid.msg_uid)
+                .filter_by(
+                    ImapUid.account_id == self.account_id,
+                    ImapUid.folder_id == self.folder_id,
+                )
+                .with_hint(
+                    ImapUid, "FORCE INDEX(ix_imapuid_account_id_folder_id_msg_uid_desc)"
```

Comment:
Here this is a covering index since we filter by account_id and folder_id and then retrieve msg_uid. Gets faster.